### DISCUSSION TITLE
How to get 'edit' functionality in V2?

### DISCUSSION THREAD

**Discussion by baartch** [TexteaInc/json-viewer#89] [Q&A]
Hi everybody

I'm unable to get any edit functionality in V2. May someone would like to help me with this?
In V1 I can add props `onAdd, onEdit, onDelete` and pass a function to these. How to achieve this in V2?
I tried `onChange` and `editable`.

```html
<JsonViewer onChange={(path) => {console.log(path)}} />

<JsonViewer onChange={(path) => {console.log(path)}} editable={true}/>
```

Thanks

**Comment by chadallard** [comment]
I have not tried this myself, but you should have had it with "editable" as this is what comes up as a parameter in the code:

`    <JsonViewer
      value={json}
      theme={ocean}
      editable={true}
    />`
    
    EDIT: I just tried this and the value doesn't seem to work at the moment. Perhaps temporarily broken feature.

**Reply by chadallard** [reply]
Answer is this is a bug:
https://github.com/TexteaInc/json-viewer/issues/92

**Comment by himself65** [comment]
We don’t have full support for editing value on our v2 like array and object edit. but you can design yourself editor by using dataTypes.
Sorry about that, we are planning to fully implement this feature, but we lack the people to work on this :(

**Reply by himself65** [reply]
The reason is our json viewer is mainly designed for the data viewer, not a editor. So this is not a top priority plan

**Reply by himself65** [reply]
But you’re welcome to contribute to this. Or if you have better idea and welcome to share!
